src/WebJobs.Script.WebHost/Controllers/KeysController.cs

```diff
@@ -0,0 +1,158 @@
+        [HttpPost]
+        [Route("admin/host/keys/{keyName}")]
+        public IHttpActionResult Post(string keyName) => AddOrUpdateFunctionSecret(keyName, null);
+
+        [HttpPut]
+        [Route("admin/functions/{name}/keys/{keyName}")]
+        public IHttpActionResult Put(string name, string keyName, Key key) => PutKey(keyName, key, name);
+
+        [HttpPut]
+        [Route("admin/host/keys/{keyName}")]
+        public IHttpActionResult Put(string keyName, Key key) => PutKey(keyName, key);
+
+        [HttpDelete]
+        [Route("admin/functions/{name}/keys/{keyName}")]
+        public IHttpActionResult Delete(string name, string keyName) => DeleteFunctionSecret(keyName, name);
+
+        [HttpDelete]
+        [Route("admin/host/keys/{keyName}")]
+        public IHttpActionResult Delete(string keyName) => DeleteFunctionSecret(keyName);
+
+        private IHttpActionResult GetKeysResult(IDictionary<string, string> keys)
+        {
+            var keysContent = new
+            {
+                keys = keys.Select(k => new { name = k.Key, value = k.Value })
+            };
+
+            var keyResponse = ApiModelUtility.CreateApiModel(keysContent, Request);
+
+            return Ok(keyResponse);
+        }
+
+        private IHttpActionResult PutKey(string keyName, Key key, string functionName = null)
+        {
+            if (functionName != null &&
+                !_scriptHostManager.Instance.Functions.Any(f => string.Equals(f.Name, functionName, StringComparison.OrdinalIgnoreCase)))
+            {
+                return NotFound();
+            }
+
+            if (key?.Value == null)
+            {
+                return BadRequest("Invalid key value");
+            }
+
+            return AddOrUpdateFunctionSecret(keyName, key.Value, functionName);
+        }
+
+        private IHttpActionResult AddOrUpdateFunctionSecret(string keyName, string value, string functionName = null)
+        {
+            if (functionName != null &&
+                !_scriptHostManager.Instance.Functions.Any(f => string.Equals(f.Name, functionName, StringComparison.OrdinalIgnoreCase)))
+            {
+                return NotFound();
+            }
+
+            KeyOperationResult operationResult;
+            if (functionName == null && string.Equals(keyName, MasterKeyName, StringComparison.OrdinalIgnoreCase))
+            {
+                operationResult = _secretManager.SetMasterKey(value);
+            }
+            else
+            {
+                operationResult = _secretManager.AddOrUpdateFunctionSecret(keyName, value, functionName);
+            }
+
+            switch (operationResult.Result)
+            {
+                case OperationResult.Created:
+                    {
+                        var keyResponse = ApiModelUtility.CreateApiModel(new { name = keyName, value = operationResult.Secret }, Request);
+                        return Created(ApiModelUtility.GetBaseUri(Request), keyResponse);
+                    }
```

Comment:
Another doc update -- POST says it'll return a 200. But looks like it'll do 200 or 201. May want to show both there.